.evergreen/csfle/kms_failpoint_server.py

```diff
@@ -0,0 +1,155 @@
+import argparse
+import http.server
+import json
+import urllib
+import ssl
+import base64
+
+"""
+Example of setting a network failpoint:
+
+$ curl -X POST localhost:3000
+{"message": "ok"}
+
+$ curl -X GET localhost:3000/set_failpoint/network/1
+{"message": "failpoint set for type: 'network'"}
+
+$ curl -X POST localhost:3000
+curl: (52) Empty reply from server
+
+$ curl -X POST localhost:3000
+{"message": "ok"}
+
+"""
+
+# A new instance of Hander is created for every request, so these have to be global variables
+failpoint_type = None
+remaining_decrypt_fails = 0
+remaining_network_fails = 0
+
+fake_ciphertext = 'a' * 96
+fake_plaintext = 'b' * 96
+
+class HTTPServerWithTLS(http.server.HTTPServer):
+    def __init__(self, server_address, Handler):
+        super().__init__(server_address, Handler)
+        self.socket = ssl.wrap_socket(
+            self.socket,
+            server_side=True,
+            certfile='../x509gen/server.pem',
+            ca_certs='../x509gen/ca.pem',
```

Comment:
This relies on the cwd being correct, I assume? I recommend either parameterizing this or getting the location of these files relative to the `__file__` global.